src/jshint.js

```diff
@@ -3754,18 +3754,25 @@ var JSHINT = (function() {
     var b;
 
     function doCatch() {
-      var e;
+      var params = [];
 
       advance("catch");
       advance("(");
 
-      e = state.tokens.next;
-      if (e.type !== "(identifier)") {
-        warning("E030", state.tokens.next, e.value);
-        e = null;
+      if (checkPunctuators(state.tokens.next, ["[", "{"])) {
+        var tokens = destructuringExpression();
+        var t;
+        for (t in tokens) {
+          t = tokens[t];
+          if (t.id) {
+            params.push({ id: t.id, token: t });
+          }
+        }
+      } else if (state.tokens.next.type !== "(identifier)") {
+        warning("E030", state.tokens.next, state.tokens.next.value);
       } else {
         // only advance if we have an identifier so we can continue parsing in the most common error - that no param is given.
-        advance();
+        params.push({ id: identifier(), token: state.tokens.curr, type: "exception" });
```

Comment:
> using identifier does allow the rest operator

Rest operator allowed because of lines 3771 and 3772.

---


```js
// jshint esnext: true
try {} catch (...error) {}
```

```
$ bin/jshint --verbose test.js
test.js: line 2, col 15, Expected an identifier and instead saw '...'. (E030)
test.js: line 2, col 15, Expected ')' and instead saw '...'. (W116)
test.js: line 2, col 18, Expected '{' and instead saw 'error'. (E021)
test.js: line 2, col 18, Expected an assignment or function call and instead saw an expression. (W030)
test.js: line 2, col 23, Missing semicolon. (W033)
test.js: line 2, col 23, Expected an identifier and instead saw ')'. (E030)
test.js: line 2, col 23, Expected an assignment or function call and instead saw an expression. (W030)
test.js: line 2, col 24, Missing semicolon. (W033)
test.js: line 2, col 18, 'error' is not defined. (W117)

9 errors
```